app/assets/javascripts/surveys.js

```diff
@@ -53,6 +53,9 @@
 
       var $surveyLink = $('#take-survey');
       var surveyUrl = survey.url;
+      if (typeof surveyUrl === "function") {
+        surveyUrl = surveyUrl();
+      }
```

Comment:
Is this required? The href="javascript:" should mean the expression is executed?